src/compiler/checker.ts

```diff
@@ -6009,8 +6009,8 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             if (!nodeIsSynthesized(range) && !(range.flags & NodeFlags.Synthesized) && (!context.enclosingFile || context.enclosingFile !== getSourceFileOfNode(range))) {
                 range = factory.cloneNode(range);
             }
-            if (!location) {
-                return range;
+            if (!location || nodeIsSynthesized(location)) {
+                return setOriginalNode(range, location?.original);

```

Comment:
If `range` already has a `.original` set, this is going to unset it if `location` is `undefined` or lacks a `.original` of its own, which, while is what fixes the issue you witness, doesn't seem desirable. Unsetting `.original` on a copy is probably not a good thing. If this issue is that when we clone a node from the cache, we `.original` it, we should update the underlining requirements in `nodeIsFullySynthetic` in `typeWriter.ts` to instead of just checking for a `.original`, specifically look for a `getParseTreeNode` instead. So,
```ts
function nodeIsFullySynthetic(node: ts.Node) {
    return ts.nodeIsSynthesized(node) && !ts.getParseTreeNode(node);
}
```

Reply:
That also works. My thought was that the node you get should be the same regardless of if it comes from the cache or from the original operation.

Reply:
Changed to using the parse tree node.